tests/test_hdf.py

```diff
@@ -0,0 +1,73 @@
+"""Tests for reading and writing hdf5 files"""
+import os
+from datetime import datetime
+import h5py
+import numpy as np
+from hdf_util import copy_hdf, create_hydrograph_times, format_date_string_hydrograph_attrib, copy_attrib, update_hydrograph
+from dss_util import read_dss_timeseries
+from fs_util import get_file
+
+
+def test_copy_hdf():
+    """Test copying HDF file and removing groups"""
+    src_file = "tests/data/Muncie.p04.hdf"
+    temp_file1 = get_file(ext=".hdf")
+    temp_file2 = get_file(ext=".hdf")
+    copy_hdf(src_file, temp_file1, ["Results"])
+    copy_hdf(src_file, temp_file2)
+    with h5py.File(temp_file1, 'r') as file:
+        n_rows = 0
+        for group1, group2 in zip(file.keys(), ["Event Conditions", "Geometry", "Plan Data"]):
+            assert group1 == group2
+            n_rows += 1
+        assert n_rows == 3
+    with h5py.File(temp_file2, 'r') as file:
+        n_rows = 0
+        for group1, group2 in zip(file.keys(), ["Event Conditions", "Geometry", "Plan Data", "Results"]):
+            assert group1 == group2
+            n_rows += 1
+        assert n_rows == 4
+    os.remove(temp_file1)
```

Comment:
Advise using `pytest` features enable a proper "teardown" where `temp_file1` and `temp_file2` are deleted after the test even if the test fails.